Automattic/wp-calypso
# Adds react and react-dom dependency to package root.

#### Changes proposed in this Pull Request

Adds `react` and `react-dom` to the root `package.json`.

This suppress lots of warnings about missing peerDependencies. There is the full list of packages that are now satisfied:

<details>
<summary>List of dependencies</summary>

```
warning " > enzyme-adapter-react-16@1.15.1" has unmet peer dependency "react@^16.0.0-0".
warning " > enzyme-adapter-react-16@1.15.1" has unmet peer dependency "react-dom@^16.0.0-0".
warning "@automattic/calypso-build > enzyme-adapter-react-16 > enzyme-adapter-utils@1.13.0" has unmet peer dependency "react@0.13.x || 0.14.x || ^15.0.0-0 || ^16.0.0-0".
warning " > react-test-renderer@16.12.0" has unmet peer dependency "react@^16.0.0".
warning "@automattic/calypso-build > jest-enzyme > jest-environment-enzyme@7.1.2" has unmet peer dependency "react@^0.13.0 || ^0.14.0 || ^15.0.0 || >=16.x".
warning "@automattic/calypso-build > enzyme-adapter-react-16 > enzyme-adapter-utils > airbnb-prop-types@2.15.0" has unmet peer dependency "react@^0.14 || ^15.0.0 || ^16.0.0-alpha".
warning " > @automattic/components@1.0.0-alpha.1" has unmet peer dependency "react@^16.8".
warning " > @automattic/components@1.0.0-alpha.1" has unmet peer dependency "react-dom@^16.8".
warning "@automattic/components > gridicons@3.3.1" has unmet peer dependency "react@^15.3.0 || ^16.0.0 ".
warning "@automattic/components > react-modal@3.11.1" has unmet peer dependency "react@^0.14.0 || ^15.0.0 || ^16".
warning "@automattic/components > react-modal@3.11.1" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0 || ^16".
warning " > @automattic/composite-checkout@1.0.0" has unmet peer dependency "react@^16.8".
warning "@automattic/composite-checkout > @emotion/core@10.0.27" has unmet peer dependency "react@>=16.3.0".
warning "@automattic/composite-checkout > @emotion/styled@10.0.27" has unmet peer dependency "react@>=16.3.0".
warning "@automattic/composite-checkout > emotion-theming@10.0.27" has unmet peer dependency "react@>=16.3.0".
warning "@automattic/composite-checkout > react-stripe-elements@4.0.2" has unmet peer dependency "react@^15.5.4 || ^16.0.0-0".
warning "@automattic/composite-checkout > react-stripe-elements@4.0.2" has unmet peer dependency "react-dom@^15.5.4 || ^16.0.0-0".
warning "@automattic/composite-checkout > @emotion/styled > @emotion/styled-base@10.0.27" has unmet peer dependency "react@>=16.3.0".
warning "@automattic/data-stores > @wordpress/data-controls > @wordpress/data > use-memo-one@1.1.1" has unmet peer dependency "react@^16.8.0".
warning "@automattic/data-stores > @wordpress/data-controls > @wordpress/data > @wordpress/compose > react-resize-aware@3.0.0" has unmet peer dependency "react@^16.8.0".
warning " > i18n-calypso@5.0.0" has unmet peer dependency "react@^16.8".
warning "@automattic/full-site-editing > @wordpress/block-editor > react-autosize-textarea@3.0.3" has unmet peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@automattic/full-site-editing > @wordpress/block-editor > react-autosize-textarea@3.0.3" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@automattic/full-site-editing > @wordpress/components > downshift@4.1.0" has unmet peer dependency "react@>=0.14.9".
warning "@automattic/full-site-editing > @wordpress/components > react-dates@17.2.0" has unmet peer dependency "react@^0.14 || ^15.5.4 || ^16.1.1".
warning "@automattic/full-site-editing > @wordpress/components > react-dates@17.2.0" has unmet peer dependency "react-dom@^0.14 || ^15.5.4 || ^16.1.1".
warning "@automattic/full-site-editing > @wordpress/components > reakit@1.0.0-beta.16" has unmet peer dependency "react@^16.8.0".
warning "@automattic/full-site-editing > @wordpress/components > reakit@1.0.0-beta.16" has unmet peer dependency "react-dom@^16.8.0".
warning "@automattic/full-site-editing > @wordpress/components > @emotion/native > @emotion/primitives-core@10.0.27" has unmet peer dependency "react@>=16.3.0".
warning "@automattic/full-site-editing > @wordpress/components > react-dates > react-outside-click-handler@1.3.0" has unmet peer dependency "react@^0.14 || >=15".
warning "@automattic/full-site-editing > @wordpress/components > react-dates > react-outside-click-handler@1.3.0" has unmet peer dependency "react-dom@^0.14 || >=15".
warning "@automattic/full-site-editing > @wordpress/components > react-dates > react-portal@4.2.1" has unmet peer dependency "react@^15.0.0-0 || ^16.0.0-0 || ^17.0.0-0".
warning "@automattic/full-site-editing > @wordpress/components > react-dates > react-with-styles@3.2.3" has unmet peer dependency "react@>=0.14".
warning "@automattic/full-site-editing > @wordpress/components > reakit > reakit-system@0.9.0" has unmet peer dependency "react@^16.8.0".
warning "@automattic/full-site-editing > @wordpress/components > reakit > reakit-system@0.9.0" has unmet peer dependency "react-dom@^16.8.0".
warning "@automattic/full-site-editing > @wordpress/components > reakit > reakit-utils@0.9.0" has unmet peer dependency "react@^16.8.0".
warning "@automattic/full-site-editing > @wordpress/components > reakit > reakit-utils@0.9.0" has unmet peer dependency "react-dom@^16.8.0".
warning "@automattic/full-site-editing > @wordpress/components > react-dates > react-with-styles > react-with-direction@1.3.1" has unmet peer dependency "react@^0.14 || ^15 || ^16".
warning "@automattic/full-site-editing > @wordpress/components > react-dates > react-with-styles > react-with-direction@1.3.1" has unmet peer dependency "react-dom@^0.14 || ^15 || ^16".
warning " > @automattic/react-i18n@1.0.0-alpha.0" has unmet peer dependency "react@^16.8".
warning " > @automattic/viewport-react@1.0.0" has unmet peer dependency "react@^16.0.0".
warning "@storybook/addon-actions > @storybook/components@5.3.14" has unmet peer dependency "react-dom@*".
warning "@storybook/addon-actions > @storybook/theming@5.3.14" has unmet peer dependency "react-dom@*".
warning "@storybook/addon-actions > @storybook/api > @reach/router@1.3.3" has unmet peer dependency "react-dom@15.x || 16.x || 16.4.0-alpha.0911da3".
warning "@storybook/addon-actions > @storybook/api > @storybook/router@5.3.14" has unmet peer dependency "react-dom@*".
warning " > @storybook/react@5.3.14" has unmet peer dependency "react@*".
warning " > @storybook/react@5.3.14" has unmet peer dependency "react-dom@*".
warning "@storybook/react > @storybook/core@5.3.14" has unmet peer dependency "react@*".
warning "@storybook/react > @storybook/core@5.3.14" has unmet peer dependency "react-dom@*".
warning " > @testing-library/react@9.3.3" has unmet peer dependency "react@*".
warning " > @testing-library/react@9.3.3" has unmet peer dependency "react-dom@*".
warning "@types/wordpress__block-editor > react-autosize-textarea@7.0.0" has unmet peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning "@types/wordpress__block-editor > react-autosize-textarea@7.0.0" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0 || ^16.0.0".
warning " > chai-enzyme@1.0.0-beta.1" has unmet peer dependency "react@^0.14.0 || ^15.0.0-0 || ^16.0.0-0".
warning " > chai-enzyme@1.0.0-beta.1" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0-0 || ^16.0.0-0".
```

</details>



#### Testing instructions

* Verify tests pass.
* Very that there are no new packages added (check `yarn.lock`).
* Checkout the branch and install deps: check that  `react` versions has not changed (`cat node_modules/react/package.json | grep version` is the same in this branch and master). Do the same for `react-dom`.
